core/serverutil.go

```diff
@@ -103,11 +103,13 @@ func taskToShortTaskInfo(t *task.Task) (sti *pb.ShortTaskInfo) {
 			OfferId: t.GetOfferId(),
 			ExecutorId: t.GetExecutorId(),
 		},
+		Pid: "UNKNOWN",
 	}
 	parentRole, ok := t.GetParentRole().(workflow.Role)
```

Comment:
I suspect that if you type-assert this one to `taskRole` instead of `workflow.Role`, you can have `GetTaskPID()` only in `taskRole`. Alternatively, perhaps `Pid` shouldn't be a member of any role at all, it could come directly from the `task.Task`. Intuitively this makes sense for me, because a PID is not a variable associated with a logical role, it is a fundamental trait of a running `Task`.